File: jimmy/courses/cpp/base/intToBin.cc
#include <iostream>
using namespace std;

const string BIN = "01";

string intToBin(int x) {
  string binStr("");
  while (x > 0) {
    binStr = BIN[x % 2] + binStr;
    x /= 2;
  }
  return binStr;
}

int main() {
  string bin = intToBin(2748);
  cout << bin << endl;
  bin = intToBin(1024);
  cout << bin << endl;
  return 0;
}

File: jimmy/courses/java/base/IntToBin.java
class BaseConverter {
  public static String intToBin(int x) {
    String binStr = "";
    while (x > 0) {
      binStr = x % 2 + binStr;
      x /= 2;
    }
    return binStr;
  }
}

public class IntToBin {
  public static void main(String[] args) {
    String bin = BaseConverter.intToBin(2748);
    System.out.println(bin);
    bin = BaseConverter.intToBin(1024);
    System.out.println(bin);
  }
}